pkg/nsx/services/subnet/subnet.go

```diff
@@ -459,3 +461,22 @@ func (service *SubnetService) UpdateSubnetSet(ns string, vpcSubnets []*model.Vpc
 	}
 	return nil
 }
+
+// GetSharedSubnetPath generates Subnet Path for shared Subnet
+func (service *SubnetService) GetSharedSubnetPath(subnet *v1alpha1.Subnet) (string, error) {
+	anno := subnet.GetAnnotations()
+	associatedResource, ok := anno[common.AnnotationAssociatedResource]
```

Comment:
the annotation was checked before "GetSharedSubnetPath" func is invoked. I feel we only need to pass the associatedResource annotation in this func.

Reply:
Updated, thanks!